src/plugins/saved_objects_management/public/services/columns/share_saved_objects_to_space_column.tsx

```diff
@@ -6,49 +6,191 @@
 
-import React, { useMemo } from 'react';
+import React, { useMemo, useState } from 'react';
 
 import { i18n } from '@kbn/i18n';
 
-import type { SpaceListProps, SpacesApiUi } from '../../../../../../x-pack/plugins/spaces/public';
-import type { SavedObjectsManagementColumn } from '../types';
+import type { SavedObjectsNamespaceType } from 'src/core/public';
+import { EuiIconTip } from '@elastic/eui';
+
+import type {
+  ShareToSpaceFlyoutProps,
+  SpaceListProps,
+  SpacesApiUi,
+} from '../../../../../../x-pack/plugins/spaces/public';
+import type { SavedObjectsManagementRecord } from '../types';
+import { SavedObjectsManagementColumn } from '../types';
 
 interface WrapperProps {
+  objectType: string;
+  objectNamespaceType: SavedObjectsNamespaceType;
   spacesApiUi: SpacesApiUi;
-  props: SpaceListProps;
+  spaceListProps: SpaceListProps;
+  flyoutProps: ShareToSpaceFlyoutProps;
 }
 
-const Wrapper = ({ spacesApiUi, props }: WrapperProps) => {
-  const LazyComponent = useMemo(() => spacesApiUi.components.getSpaceList, [spacesApiUi]);
+const columnName = i18n.translate('savedObjectsManagement.shareToSpace.columnTitle', {
+  defaultMessage: 'Spaces',
+});
+const columnDescription = i18n.translate('savedObjectsManagement.shareToSpace.columnDescription', {
+  defaultMessage: 'The spaces that this object is currently assigned to',
+});
+const isolatedObjectTypeTitle = i18n.translate(
+  'savedObjectsManagement.shareToSpace.isolatedObjectTypeTitle',
+  { defaultMessage: 'Isolated object type' }
+);
+const isolatedObjectTypeContent = i18n.translate(
+  'savedObjectsManagement.shareToSpace.isolatedObjectTypeContent',
+  {
+    defaultMessage:
+      'This saved object type only exists in one space, it cannot be assigned to multiple spaces.',
+  }
+);
+const shareableSoonObjectTypeContent = i18n.translate(
+  'savedObjectsManagement.shareToSpace.shareableSoonObjectTypeContent',
+  {
+    defaultMessage:
+      'Coming soon: this saved object type only exists in one space, but it will be assignable to multiple spaces in a future release.',
+  }
+);
+const globalObjectTypeTitle = i18n.translate(
+  'savedObjectsManagement.shareToSpace.globalObjectTypeTitle',
+  { defaultMessage: 'Global object type' }
+);
+const globalObjectTypeContent = i18n.translate(
+  'savedObjectsManagement.shareToSpace.globalObjectTypeContent',
+  { defaultMessage: 'This saved object type always exists in all spaces.' }
+);
+
+/**
+ * This is a hard-coded list that can be removed when each of these "share-capable" object types are made to be shareable.
+ * Note, this list does not preclude other object types from being made shareable in the future, it just consists of the object types that
+ * we are working towards making shareable in the near term.
+ */
+const SHAREABLE_SOON_OBJECT_TYPES = [
+  'tag',
+  'dashboard',
+  'canvas-workpad',
+  'canvas-element',
+  'lens',
+  'visualization',
+  'map',
+  'graph-workspace',
+  'search',
+  'query',
+  'rule',
+  'connector',
+];
```

Comment:
I needed some way to identify which object types we are planning to make shareable in the near term, so we could show a different tooltip for those ("Coming soon: ...").

This hard-coded list isn't ideal, but it works for our simple use case here. Once these object types are made shareable, the list will be obsolete and we can remove it.

Reply:
A hard-coded list is a pain to keep updated. Ideally, the list of "share-capable" saved object types should exist closer to a registry of which saved objects can be sharable. That way, when they do change to be shareable, the list will automatically update.

Where that should go, I'm not sure though, and open to suggestions.

For now, keeping it as a list should be ok, although I'd _prefer_ to have it in a constants folder/file.

Reply:
> A hard-coded list is a pain to keep updated. Ideally, the list of "share-capable" saved object types should exist closer to a registry of which saved objects can be sharable. That way, when they do change to be shareable, the list will automatically update.

Yeah, in principle I would agree. That would need to take place in the saved object type registration (where each object type is defined, along with whether it is shareable or not).

But then we would need to make changes to lots of different plugins, and we would have to change the SOM server code to return this field to the client side code as extra metadata, it gets messy really quickly for what amounts to a small UI tweak on a single page.

Glad you are OK with this list for now, I'm hoping we can remove it completely in the next few releases. I moved it to a constants file just now: 13b6d92fdfa581362226ec8734f7bffc18392188